internal/controller/promotions/promotions.go

```diff
@@ -185,8 +185,6 @@ func (r *reconciler) Reconcile(
 		return ctrl.Result{}, fmt.Errorf("error initializing Promotion queues: %w", err)
 	}
 
-	ctx = logging.ContextWithLogger(ctx, logger)
-
```

Comment:
Good eye @devholic. As written, this is indeed redundant...

It was _supposed_ to look like this:

```
logger = logging.LoggerFromContext(ctx).WithFields(log.Fields{
	"namespace": req.NamespacedName.Namespace,
	"promotion": req.NamespacedName.Name,
})
ctx = logging.ContextWithLogger(ctx, logger)
logger.Debug("reconciling Promotion")
```

Instead of removing this line, could we add this additional information? It's valuable context for all other log messages produced by the various functions that this one will call into.

Reply:
Note that all the other reconcilers have lines like these. It's a mistake that they were absent here.

Reply:
It would be much better :) Followed your suggestion in https://github.com/akuity/kargo/pull/1821/commits/b6099de10a5d262867f35b6c575efe15b2eb4b11 (and updated PR title/desc too)

Reply:
Thank you!